types/aggregator.go

```diff
@@ -67,35 +67,35 @@ type AggregatorMetadata struct {
 // MetadataK8S See: https://github.com/kubernetes/website/blob/60390ff3c0ef0043a58568ad2e4c2b7634028074/content/en/examples/pods/inject/dapi-volume.yaml#L5
 // For further cluster information data that can be included check: https://pkg.go.dev/k8s.io/client-go/discovery#DiscoveryClient.
 type MetadataK8S struct {
-	Namespace       string `json:"k8s.namespace"`
```

Comment:
From what I gather the intention is to embed all these into the `AggregatorMetadata` struct and then only have the relevant fields show up. If so, this is a bit of an abuse of how the language works. I can concede to it since I know no other better mechanism in go since there is no `union` type or the really bizarre `Or` type from haskell.

Can you add a comment to `AggregatorMetadata` to clarify that intent?

Reply:
Done.
@pwhelan